main/src/ca/uwaterloo/flix/language/errors/WeederError.scala

```diff
@@ -772,4 +772,47 @@ object WeederError {
     })
 
   }
+
+  /**
+    * An error raised to indicate an illegal ascription on a formal parameter.
+    *
+    * @param loc the location where the error occurred.
+    */
+  case class IllegalFormalParamAscription(loc: SourceLocation) extends WeederError {
+    def summary: String = "Unexpected type ascription. Type ascriptions are not permitted on effect handler cases."
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> Unexpected type ascription. Type ascriptions are not permitted on effect handler cases.
```

Comment:
Maybe instead of ascription we should say signature? Or is that more confusing.

Reply:
Seems like we would be overloading the term too much. But I agree that "ascription" feels a little technical/awkward.